### PR TITLE
Add python wrapper for multiplex operator.

### LINKED ISSUE #7717: Need add python wrapper for multiplex operator

### PR DESCRIPTION
**Pull request by pkuyym**
Resolves #7717

### CODE REVIEW

File: python/paddle/v2/fluid/layers/nn.py [hunk 1]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+    variables to output a multiplex variable. Assuming that there are :math:`m`
+    input variables and let :math:`I_i` represents the i-th input variable and i
+    is in [0, :math:`m`). All input variables are tensors with same shape
+    [:math:`d_0`, :math:`d_1`, ..., :math:`d_R`]. Please note that rank of the
+    input tensor should be at least 2. Each input variable will be viewed as a
+    2-D matrix with shape [:math:`M`, :math:`N`] where :math:`M` for :math:`d_0`
+    and :math:`N` for :math:`d_1` * :math:`d_2` * ... * :math:`d_R`. Let
+    :math:`I_i[j]` be the j-th row of the i-th input variable. The given index
+    variable should be a 2-D tensor with shape [:math:`M`, 1]. Let `ID[i]` be
+    the i-th index value of index variable. Then the output variable will be a
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```the i-th index value of index variable``` -> ```the i-th index of the index variable```

File: python/paddle/v2/fluid/layers/nn.py [hunk 2]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+    tensor with shape [:math:`d_0`, :math:`d_1`, ..., :math:`d_R`]. If we view
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```view``` -> ```treat```

File: python/paddle/v2/fluid/layers/nn.py [hunk 3]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+    the output tensor as a 2-D matrix with shape [:math:`M`, :math:`N`] and let
+    :math:`O[i]` be the i-th row of the matrix, then values of `O[i]` come from
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```then values of O[i] come from``` -> ```then O[i] is equal to```

File: python/paddle/v2/fluid/layers/nn.py [hunk 4]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+    :math:`I_{ID[i]}[i]`.
+
+    Args:
+       inputs (list): Input variables which are tensors with same shape and the
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```Input variables which are tensors with same shape``` -> ```A variable list in which every variable has the same shape```

File: python/paddle/v2/fluid/layers/nn.py [hunk 5]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+                rank is at least 2.
+       index (Variable): Tensor<int32>, index variable which is a 2-D tensor
+                with shape [M, 1] where M for batch size.
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```where M for batch size``` -> ```where M is the batch size```

File: python/paddle/v2/fluid/layers/nn.py [hunk 6]
```diff
@@ -2618,3 +2619,50 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
+def multiplex(inputs, index):
+    """
+    **Multiplex Layer**
+
+    Referring to the given index variable, this layer gathers from the input
+    variables to output a multiplex variable. Assuming that there are :math:`m`
+    input variables and let :math:`I_i` represents the i-th input variable and i
+    is in [0, :math:`m`). All input variables are tensors with same shape
+    [:math:`d_0`, :math:`d_1`, ..., :math:`d_R`]. Please note that rank of the
+    input tensor should be at least 2. Each input variable will be viewed as a
```
[the remark this answers is not in the document]
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 1 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 2 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 3 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done. Thanks.

File: python/paddle/v2/fluid/layers/nn.py (hunk 4 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 5 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py [hunk 7]
```diff
@@ -2707,3 +2708,55 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
...
     return helper.append_activation(out)
+
+
+def multiplex(inputs, index):
+    """
+    **Multiplex Layer**
+
+    Referring to the given index variable, this layer gathers from the input
+    variables to output a multiplex variable. Assuming that there are :math:`m`
+    input variables and let :math:`I_i` represents the i-th input variable and i
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```represents```->```represent``` or delete ```let```

File: python/paddle/v2/fluid/layers/nn.py [hunk 8]
```diff
@@ -2707,3 +2708,55 @@ def row_conv(input, future_context_size, param_attr=None, act=None):
                 'Filter': [filter_param]},
         outputs={'Out': [out]})
     return helper.append_activation(out)
+
+
+def multiplex(inputs, index):
+    """
+    **Multiplex Layer**
+
+    Referring to the given index variable, this layer gathers from the input
```
Comment (kuke) [inline python/paddle/v2/fluid/layers/nn.py]: ```this layer gathers from the input variables to output a multiplex variable``` -> ```this layer fetches rows from the input variables and constructs a multiplex variable for output```

File: python/paddle/v2/fluid/layers/nn.py (hunk 7 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 8 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Thanks, done.

**Pull request merged**